src/reactpy_router/components.py

```diff
@@ -0,0 +1,101 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Any
+from urllib.parse import urljoin
+from uuid import uuid4
+
+from reactpy import component, event, html, use_connection
+from reactpy.backend.types import Location
+from reactpy.core.component import Component
+from reactpy.core.types import VdomDict
+from reactpy.web.module import export, module_from_file
+
+from reactpy_router.hooks import _use_route_state
+from reactpy_router.types import Route
+
+History = export(
+    module_from_file("reactpy-router", file=Path(__file__).parent / "static" / "bundle.js"),
+    ("History"),
+)
+"""Client-side portion of history handling"""
+
+Link = export(
+    module_from_file("reactpy-router", file=Path(__file__).parent / "static" / "bundle.js"),
+    ("Link"),
+)
+"""Client-side portion of link handling"""
+
+link_js_content = (Path(__file__).parent / "static" / "link.js").read_text(encoding="utf-8")
+
+
+def link(attributes: dict[str, Any], *children: Any) -> Component:
```

Comment:
Change `attributes` type to a `TypedDict`

Reply:
Using `TypedDict` with `total=false` was causing a bunch of issues, so I'll skip this for now.